uberenv.py

```diff
@@ -828,38 +852,6 @@ def patch(self):
         self.disable_spack_config_scopes(spack_dir)
         spack_etc_defaults_dir = pjoin(spack_dir,"etc","spack","defaults")
 
```

Comment:
@cyrush I just want you to be aware that this will no longer hot-copy any files into spack's repo. All of these things can be handled inside of the environment file.

Reply:
In that case, do we need to add safeguards/options our environment files to not use the user's home directory (e.g. `~/.spack`) ?

Reply:
I'm confused by this. Can this section be removed, then (in another PR)?

Reply:
The disabling of the user scope config (inside your home directory) is handled by the function `disable_spack_config_scopes`, which is still there.

Reply:
this made me realize that `spack_etc_defaults_dir` was no longer being used, so i removed it